File: RedSheet.DbEntities/Models/vProjectOutcomeAndLearningRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectOutcomeAndLearningRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectOutcomeAndLearningRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int OutcomeAndLearningCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectModuleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectOutcomeAndLearningId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string ProjectOutcomeAndLearningValue { get; set; } 

	public vProjectOutcomeAndLearningRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-outcome-and-learning-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectOutcomeAndLearningRecord {
    constructor(vProjectOutcomeAndLearningRecord?: vProjectOutcomeAndLearningRecord )  {
        let properties = [ "outcomeAndLearningCategoryId", "projectModuleId", "projectOutcomeAndLearningId", "projectOutcomeAndLearningValue",];
        for (let property of properties)
            if (vProjectOutcomeAndLearningRecord && vProjectOutcomeAndLearningRecord[property])
                this[property] = vProjectOutcomeAndLearningRecord[property];
    }
 
    @range(1,2147483647)
	outcomeAndLearningCategoryId : number =   undefined;
 
	projectModuleId : number =   0 ;
 
    @range(1,2147483647)
	projectOutcomeAndLearningId : number =   undefined;
 
    @required()
    @maxLength(1000)
	projectOutcomeAndLearningValue : string =   undefined;


}
